Add custom notification sound for better support on Firefox

eikendev opened the pull request:
This pull request fixes #263.

We
- add a custom notification sound, licensed under CC0,
- set a priority threshold to only play the sound for *important* notifications,
- and disable default notification sounds privided by other browsers.

eikendev commented:
I've proposed a change to the type definitions for `INotifyOption` at https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42232. Once that is merged we can get rid of the cast to `any`.

jmattheis commented on ui/src/snack/browserNotification.ts:
```diff
@@ -19,10 +19,17 @@ export function notifyNewMessage(msg: IMessage) {
     notify.show();
+
+    if (msg.priority >= 4 && !Notify.needsPermission) {
+        const let src = 'static/notification.ogg';
```
Compile error
```suggestion
        const src = 'static/notification.ogg';
```

jmattheis commented on ui/src/snack/browserNotification.ts:
```diff
@@ -19,10 +19,17 @@ export function notifyNewMessage(msg: IMessage) {
+
+    if (msg.priority >= 4 && !Notify.needsPermission) {
+        const let src = 'static/notification.ogg';
+        const let audio = new Audio(src);
```
Compile error
```suggestion
        const audio = new Audio(src);
```

jmattheis commented on ui/src/snack/browserNotification.ts:
```diff
@@ -19,10 +19,17 @@ export function notifyNewMessage(msg: IMessage) {
     const notify = new Notify(msg.title, {
         body: removeMarkdown(msg.message),
         icon: msg.image,
+        silent: true,
```
The pr was merged, could you reference the newer version of the dependency?

jmattheis commented on ui/src/snack/browserNotification.ts:
```diff
@@ -19,10 +19,17 @@ export function notifyNewMessage(msg: IMessage) {
+    } as any);
     notify.show();
+
+    if (msg.priority >= 4 && !Notify.needsPermission) {
```
I think the audio should be played regardless if the user allows notifications. Furthermore, I think the audio should be started inside the reaction.ts
```typescript
stores.wsStore.listen((message) => {
    stores.messagesStore.publishSingleMessage(message);
    Notifications.notifyNewMessage(message);
    if (message.priority >= 4) {
        const src = 'static/notification.ogg';
        const audio = new Audio(src);
        audio.play();
    }
});
```

eikendev commented:
Sorry, I was sure that I tested d954baa9535e3202ca5c5285fb7ce74b2e10f254, turns out I was wrong. I've now upgraded `@types/notifyjs` to the latest version and removed the cast. Also, I moved the block into `reactions.ts` as suggested.
